README.md
Demo.preseed.in
===============


lllbe/CHANGELOG.md
# Headhesive changelog

## v1.2.4
- Fix resize event when using numerical offset
- Fix destroy method for resize event
- Changelog corrections

## v1.2.3
- Readme correction

## v1.2.2
- Fix UMD support

## v1.2.1
- Bump version to publish to npm

## v1.2.0
- Add UMD support
- Allow to select either the top or bottom of the offset element
- Update the offset position on browser resize
- Publish to npm
- Add license file
- Make the clone version lower z-index than original. Prevents visual overlap of both when scrolling quickly

## v1.1.1
- Fix class naming

## v1.1.0
- Call the init method internally on instance creation
- Destroy method to only destroy it's instance

## v1.0.0
- Initial


lllbe/js/main.js
jQuery(document).ready(function($){
	//set animation timing
	var animationDelay = 2500,
		//loading bar effect
		barAnimationDelay = 3800,
		barWaiting = barAnimationDelay - 3000, //3000 is the duration of the transition on the loading bar - set in the scss/css file
		//letters effect
		lettersDelay = 50,
		//type effect
		typeLettersDelay = 150,
		selectionDuration = 500,
		typeAnimationDelay = selectionDuration + 800,
		//clip effect 
		revealDuration = 600,
		revealAnimationDelay = 1500;
	
	initHeadline();
	autoPlayYouTubeModal();
	//FUNCTION TO GET AND AUTO PLAY YOUTUBE VIDEO FROM DATATAG
  function autoPlayYouTubeModal() {
      var trigger = $("body").find('[data-toggle="modal"]');
      trigger.click(function () {
          var theModal = $(this).data("target"),
              videoSRC = $(this).attr("data-theVideo"),
              videoSRCauto = videoSRC + "?autoplay=1";
          $(theModal + ' iframe').attr('src', videoSRCauto);
          $(theModal + ' button.close').click(function () {
              $(theModal + ' iframe').attr('src', videoSRC);
          });
      });
  }

	

	function initHeadline() {
		//insert <i> element for each letter of a changing word
		singleLetters($('.cd-headline.letters').find('b'));
		//initialise headline animation
		animateHeadline($('.cd-headline'));
	}

	function singleLetters($words) {
		$words.each(function(){
			var word = $(this),
				letters = word.text().split(''),
				selected = word.hasClass('is-visible');
			for (i in letters) {
				if(word.parents('.rotate-2').length > 0) letters[i] = '<em>' + letters[i] + '</em>';
				letters[i] = (selected) ? '<i class="in">' + letters[i] + '</i>': '<i>' + letters[i] + '</i>';
			}
		    var newLetters = letters.join('');
		    word.html(newLetters);
		});
	}

	

  
	function animateHeadline($headlines) {
		var duration = animationDelay;
		$headlines.each(function(){
			var headline = $(this);
			
			if(headline.hasClass('loading-bar')) {
				duration = barAnimationDelay;
				setTimeout(function(){ headline.find('.cd-words-wrapper').addClass('is-loading') }, barWaiting);
			} else if (headline.hasClass('clip')){
				var spanWrapper = headline.find('.cd-words-wrapper'),
					newWidth = spanWrapper.width() + 10
				spanWrapper.css('width', newWidth);
			} else if (!headline.hasClass('type') ) {
				//assign to .cd-words-wrapper the width of its longest word
				var words = headline.find('.cd-words-wrapper b'),
					width = 0;
				words.each(function(){
					var wordWidth = $(this).width();
				    if (wordWidth > width) width = wordWidth;
				});
				headline.find('.cd-words-wrapper').css('width', width);
			};

			//trigger animation
			setTimeout(function(){ hideWord( headline.find('.is-visible').eq(0) ) }, duration);
		});
	}

	function hideWord($word) {
		var nextWord = takeNext($word);
		
		if($word.parents('.cd-headline').hasClass('type')) {
			var parentSpan = $word.parent('.cd-words-wrapper');
			parentSpan.addClass('selected').removeClass('waiting');	
			setTimeout(function(){ 
				parentSpan.removeClass('selected'); 
				$word.removeClass('is-visible').addClass('is-hidden').children('i').removeClass('in').addClass('out');
			}, selectionDuration);
			setTimeout(function(){ showWord(nextWord, typeLettersDelay) }, typeAnimationDelay);
		
		} else if($word.parents('.cd-headline').hasClass('letters')) {
			var bool = ($word.children('i').length >= nextWord.children('i').length) ? true : false;
			hideLetter($word.find('i').eq(0), $word, bool, lettersDelay);
			showLetter(nextWord.find('i').eq(0), nextWord, bool, lettersDelay);

		}  else if($word.parents('.cd-headline').hasClass('clip')) {
			$word.parents('.cd-words-wrapper').animate({ width : '2px' }, revealDuration, function(){
				switchWord($word, nextWord);
				showWord(nextWord);
			});

		} else if ($word.parents('.cd-headline').hasClass('loading-bar')){
			$word.parents('.cd-words-wrapper').removeClass('is-loading');
			switchWord($word, nextWord);
			setTimeout(function(){ hideWord(nextWord) }, barAnimationDelay);
			setTimeout(function(){ $word.parents('.cd-words-wrapper').addClass('is-loading') }, barWaiting);

		} else {
			switchWord($word, nextWord);
			setTimeout(function(){ hideWord(nextWord) }, animationDelay);
		}
	}

	function showWord($word, $duration) {
		if($word.parents('.cd-headline').hasClass('type')) {
			showLetter($word.find('i').eq(0), $word, false, $duration);
			$word.addClass('is-visible').removeClass('is-hidden');

		}  else if($word.parents('.cd-headline').hasClass('clip')) {
			$word.parents('.cd-words-wrapper').animate({ 'width' : $word.width() + 10 }, revealDuration, function(){ 
				setTimeout(function(){ hideWord($word) }, revealAnimationDelay); 
			});
		}
	}

	function hideLetter($letter, $word, $bool, $duration) {
		$letter.removeClass('in').addClass('out');
		
		if(!$letter.is(':last-child')) {
		 	setTimeout(function(){ hideLetter($letter.next(), $word, $bool, $duration); }, $duration);  
		} else if($bool) { 
		 	setTimeout(function(){ hideWord(takeNext($word)) }, animationDelay);
		}

		if($letter.is(':last-child') && $('html').hasClass('no-csstransitions')) {
			var nextWord = takeNext($word);
			switchWord($word, nextWord);
		} 
	}

	function showLetter($letter, $word, $bool, $duration) {
		$letter.addClass('in').removeClass('out');
		
		if(!$letter.is(':last-child')) { 
			setTimeout(function(){ showLetter($letter.next(), $word, $bool, $duration); }, $duration); 
		} else { 
			if($word.parents('.cd-headline').hasClass('type')) { setTimeout(function(){ $word.parents('.cd-words-wrapper').addClass('waiting'); }, 200);}
			if(!$bool) { setTimeout(function(){ hideWord($word) }, animationDelay) }
		}
	}

	function takeNext($word) {
		return (!$word.is(':last-child')) ? $word.next() : $word.parent().children().eq(0);
	}

	function takePrev($word) {
		return (!$word.is(':first-child')) ? $word.prev() : $word.parent().children().last();
	}

	function switchWord($oldWord, $newWord) {
		$oldWord.removeClass('is-visible').addClass('is-hidden');
		$newWord.removeClass('is-hidden').addClass('is-visible');
	}

	//this is for the demo only
	var intro = $('.cd-intro');
	$('.cd-filter input').on('change', function(event){
		var selected = $(event.target).attr('id')
		switch(selected) {
			case 'rotate-1':
				intro.load('content.html .rotate-1', function(){
					initHeadline();
				});
				break;
			case 'type':
				intro.load('content.html .type', function(){
					initHeadline();
				});
				break;
			case 'rotate-2':
				intro.load('content.html .rotate-2', function(){
					initHeadline();
				});
				break;
			case 'loading-bar':
				intro.load('content.html .loading-bar', function(){
					initHeadline();
				});
				break;
			case 'slide':
				intro.load('content.html .slide', function(){
					initHeadline();
				});
				break;
			case 'clip':
				intro.load('content.html .clip', function(){
					initHeadline();
				});
				break;
			case 'zoom':
				intro.load('content.html .zoom', function(){
					initHeadline();
				});
				break;
			case 'rotate-3':
				intro.load('content.html .rotate-3', function(){
					initHeadline();
				});
				break;
			case 'scale':
				intro.load('content.html .scale', function(){
					initHeadline();
				});
				break;
			case 'push':
				intro.load('content.html .push', function(){
					initHeadline();
				});
				break;
		}
	});
});

lllbe/mediajs/common.js
/*-------------------------------------------------------------------------

	Theme Name: EGO - html - v.1
	
	For any questions concerning this theme please refer to documention or
	our forum at support.udfrance.com.

/*------------------------------------------------------------------------

//GENERAL FUNCTONS ///////////////////////////////////////////////////////

-------------------------------------------------------------------------*/

$(document).ready(function(){
						   
	
	/*vars used throughout*/
	var thumb = $('.thumb,.round-thumb'),
		thumbW,
		thumbH,
		thumbCaption,
		target,
		hoverSpeed=500,
		hoverEase='easeOutExpo',
	 	targetNetwork =$('ul.socialSmall li a'),
		toggleMenu =$('.mobileMenuToggle'),
		lightboxTransition = 'fade',				//Set lightbox transition type
	 	overlayOpacity =0.8,						//Fancybox overlay opacity
	 	overlayColor = '#000',						//Fancybox overlay color	
	 	videoWidth = 663,							//Fancybox video width
	 	videoHeight = 385;							//Fancybox video height
		lazyload = true;


	
	//LAZY LOADING -------------------------------------------------------------------------/
		

  	$(function() {
		
		 if(lazyload==false || isMobile == true) return false;
			 
          $("img.lazy").lazyload({
             placeholder : "images/blank.gif",
             effect : "fadeIn"
          });
		  
     });
	
	
	//MOBILE MENU -----------------------------------------------------------------------/
	
	
	toggleMenu.on('click', function(event) {
									
		if($(this).parent().find('ul.navigation').is(':hidden')){
		
			$(this).parent().find('ul.navigation').slideDown();
			$(this).addClass('open');

			
		}else{
			
			$(this).parent().find('ul.navigation').slideUp();
			$(this).removeClass('open');
		
			
		}
		
		event.preventDefault();
									
	});
	

	//ROLLOVER SPECIFIC ---------------------------------------------------------------------/
		
		
	/*general
	-------------------*/
			
	thumb.on({

		 mouseenter: function () {
				 
			 //check if device is mobile 
			 //or within an inactive filter category
			 //or if its video content in which case do nothing
			 if(isMobile == true) return false;
			 
			 thumbW = thumb.find('a').find('img').width();
			 thumbH = thumb.find('a').find('img').height();
			 
			//get refrences needed
		 	thumbCaption = $(this).find('a').attr('title');
			
			//add rolloverscreen
			if(!$(this).find('a').find('div').hasClass('thumb-rollover')) $(this).find('a').append('<div class="thumb-rollover"></div>');
			
			
			//set it to the image size and fade in
			hoverScreen = $('.thumb-rollover')
			hoverScreen.css({width:thumbW,height:thumbH});

			
			//make sure caption is filled out
			if (typeof thumbCaption !== 'undefined' && thumbCaption !== false && $(this).find(hoverScreen).is(':empty')) {
				
				//construct rollover & animate
   				$(this).find(hoverScreen).append('<div class="thumbInfo">'+thumbCaption+'</div>');
				target = $(this).find(hoverScreen);
				target.stop().animate({opacity:1},hoverSpeed, hoverEase);
			}
			
		}, 
		
		  mouseleave: function () {
		
			if(isMobile == true) return false;
			
			//animate out
			$(this).find(hoverScreen).animate({opacity:0},hoverSpeed,hoverEase,function(){
	
					//delete rollover
				   $(this).remove();
				
			});
			
		
		 }
	
	  });
	
	
	//LIGHTBOX SPECIFIC ---------------------------------------------------------------------/


	/*lightbox-img
	-------------------------------*/
	
	//if($('a.lightbox').length == 0) return false;
	
	$('a.lightbox').fancybox({
										   
			'transitionIn'		: lightboxTransition,
			'transitionOut'		: lightboxTransition,
			'titlePosition' 	: 'over',
			'padding'			: '0',																		
			'overlayOpacity'    : overlayOpacity,
			'overlayColor'      : overlayColor,
			'titleFormat'		: function(title, currentArray, currentIndex, currentOpts) {  
						
		
			var obj = currentArray[ currentIndex ] //get current image
			var target = $(obj).parent();		   //get its container	
		
		
			//CASE 1: thumb has associated html content
			if($(target).next().hasClass('fancybox-html')){
					
					
				//check if stack order should be displayed
									
				if ($(target).next().length && $(obj).attr('rel')){
										
						return  '<span id="fancybox-title-over">' + '<div class="fancybox-num"> Image:'+(currentIndex + 1) + ' / ' + currentArray.length+'</div>'+($(target).next().html()) + '</span>';
										
				}else {
											
						return  '<span id="fancybox-title-over">' + ($(target).next().html()) + '</span>';
										
				}
									
								
			//CASE 2:  thumb is a part of a group and has a title only
			} else if($(obj).attr('rel') && $(obj).attr('title')){
		
					return  '<span id="fancybox-title-over">' + '<div class="fancybox-num"> Image:'+ (currentIndex + 1) + ' / ' + currentArray.length + '</div> '+ (title.length?''+title:'') + '</span>';
									
								
			//CASE 3: thumb has no info but belongs to group
			} else if($(obj).attr('rel')) {
								
					return  '<span id="fancybox-title-over">' + '<div class="fancybox-num" style="margin-bottom:0px"> Image:'+(currentIndex + 1) + ' / ' + currentArray.length+'</div>'+'</span>';
								
								
			//CASE 4: thumb has a title only
			} else if($(obj).attr('title')) {
								
				//if image is not associated with group, hide image numbering
				return  '<span id="fancybox-title-over">' +(title.length ?''+title :'') + '</span>';
									
								
				//CASE 5: no info & does not belong to group
				}else{
									
				// hide title overlay
				$('#fancybox-title-over').hide();
									
				}
							
			},
						
						
				//animate image info on complete
				'onComplete':function(){
						
						$('#fancybox-title-over').hide().fadeIn('slow');
						
				}	 
						
			});
	
	
	/*lightbox-img
	-------------------------------*/
	
	$('a.media').fancybox({
								   
        'transitionIn'        : lightboxTransition,
        'transitionOut'       : lightboxTransition,
		'padding'			  : '0',	
		'titlePosition'		  : 'outside',
		'width'			      : videoWidth,
		'height'			  : videoHeight,
		'overlayOpacity'      : overlayOpacity,
		'overlayColor'        : overlayColor,
        'autoscale'           : false,
        'type'                : 'iframe',
		'swf'           	  : {
		'wmode'               : 'transparent',
		'allowfullscreen'  	: 'true'}

        }); 
	
	
	
});

lllbe/mediajs/contact.js
$(document).ready(function() {
	var contactFormDefaults=new Array();
	contactFormDefaults['name']='Your Name';
	contactFormDefaults['email']='E-mail';
	contactFormDefaults['subject']='Subject';
	contactFormDefaults['message']='Message';
	contactFormDefaults['msg']=$('.contactForm span#msg').html();
	
	$('.contactForm input[type="text"], .contactForm textarea').focus(function() {
		$(this).addClass('inputHighlight').removeClass('errorOutline');
		if($(this).hasClass('required')) {
			$('.contactForm span#msg').html('This is a required field.').removeClass('errorMsg successMsg');
		} else {
			$('.contactForm span#msg').html(contactFormDefaults['msg']).removeClass('errorMsg successMsg');
		}
		if($(this).val()==contactFormDefaults[$(this).attr('id')]) {
			$(this).val('');
		}
	});
	$('.contactForm input[type="text"], .contactForm textarea').blur(function() {
		$(this).removeClass('inputHighlight');
		$('.contactForm span#msg').html(contactFormDefaults['msg']).removeClass('errorMsg successMsg');
		if($(this).val()=='') {
			$(this).val(contactFormDefaults[$(this).attr('id')]);
		}
	});
	
	$('.contactForm input[type="text"], .contactForm textarea').hover(function() {
			$(this).addClass('inputHighlight');
		},
		function() {	
			$(this).not(':focus').removeClass('inputHighlight');
		}
	);
	
	$('.contactForm').submit(function() {
		$('.contactForm .submit').attr("disabled", "disabled");
		$('#msg').html('<img src="images/loader-light.gif" />').removeClass('errorMsg successMsg');
		var isError=false;
		$('.contactForm input, .contactForm textarea').each(function() {
			if($(this).hasClass('required') && ($.trim($(this).val())==contactFormDefaults[$(this).attr('id')] || $.trim($(this).val())=='')) {
				$(this).addClass('errorOutline');
				$('#msg').html('There was an error sending your message. Please try again.').addClass('errorMsg');
				isError=true;
			}
			if($(this).attr('id')=='email') {
				var reg = /^([A-Za-z0-9_\-\.])+\@([A-Za-z0-9_\-\.])+\.([A-Za-z]{2,4})$/;
				if(reg.test($(this).val())==false) {
					$(this).addClass('errorOutline');
					if(!isError) {
						$('#msg').html('Please enter a valid e-mail address and try again.').addClass('errorMsg');
					}
					isError=true;
				}
			}
		});
		if(isError) {
			$('.contactForm .submit').removeAttr("disabled");
			return false;
		} else {
			var name = $('#name').val(), email = $('#email').val(), subject = $('#subject').val(), message = $('#message').val();
			$.ajaxSetup ({
				cache: false
			});
			var dataString = 'name='+ name + '&email=' + email + '&subject=' + subject + '&message=' + message + '&ajax=1';
			$.ajax({
				type: "POST",
				url: "php/contact.php",
				data: dataString,
				success: function(msg) {
					// Check to see if the mail was successfully sent
					if(msg=='Mail sent') {
						// Update the progress bar
						$('#msg').html('Message sent.').addClass('successMsg');
						// Reset the subject field and message textbox
						if(contactFormDefaults['subject']) {
							$('#subject').val(contactFormDefaults['subject']);
						} else {
							$('#subject').val('');
						}
						if(contactFormDefaults['message']) {
							$('#message').val(contactFormDefaults['message']);
						} else {
							$('#message').val('');
						}
					} else {
						$('#msg').html('There was an error sending your email. Please try again.').addClass('errorMsg');
						$('.contactForm .submit').attr("disabled", "");
					}
					// Activate the submit button
					$('.contactForm .submit').removeAttr("disabled");
				},
				error: function(ob,errStr) {
					$('#msg').html('There was an error sending your email. Please try again.').addClass('errorMsg');
					//Activate the submit button
					$('.contactForm .submit').removeAttr("disabled");
				}
			});
			return false;
		}
	});
});
<!--mailchimp-->
<script type="text/javascript">
var fnames = new Array();var ftypes = new Array();fnames[0]='EMAIL';ftypes[0]='email';fnames[1]='FNAME';ftypes[1]='text';fnames[2]='LNAME';ftypes[2]='text';
try {
var jqueryLoaded=jQuery;
jqueryLoaded=true;
} catch(err) {
var jqueryLoaded=false;
}
var head= document.getElementsByTagName('head')[0];
if (!jqueryLoaded) {
var script = document.createElement('script');
script.type = 'text/javascript';
script.src = '//ajax.googleapis.com/ajax/libs/jquery/1.4.4/jquery.min.js';
head.appendChild(script);
if (script.readyState && script.onload!==null){
script.onreadystatechange= function () {
if (this.readyState == 'complete') mce_preload_check();
} 
}
}

var err_style = '';
try{
err_style = mc_custom_error_style;
} catch(e){
err_style = '#mc_embed_signup input.mce_inline_error{border-color:#6B0505;} #mc_embed_signup div.mce_inline_error{margin: 0 0 1em 0; padding: 5px 10px; background-color:#6B0505; font-weight: bold; z-index: 1; color:#fff;}';
}
var head= document.getElementsByTagName('head')[0];
var style= document.createElement('style');
style.type= 'text/css';
if (style.styleSheet) {
style.styleSheet.cssText = err_style;
} else {
style.appendChild(document.createTextNode(err_style));
}
head.appendChild(style);
setTimeout('mce_preload_check();', 250);

var mce_preload_checks = 0;
function mce_preload_check(){
if (mce_preload_checks>40) return;
mce_preload_checks++;
try {
var jqueryLoaded=jQuery;
} catch(err) {
setTimeout('mce_preload_check();', 250);
return;
}
var script = document.createElement('script');
script.type = 'text/javascript';
script.src = 'http://downloads.mailchimp.com/js/jquery.form-n-validate.js';
head.appendChild(script);
try {
var validatorLoaded=jQuery("#fake-form").validate({});
} catch(err) {
setTimeout('mce_preload_check();', 250);
return;
}
mce_init_form();
}
function mce_init_form(){
jQuery(document).ready( function($) {
var options = { errorClass: 'mce_inline_error', errorElement: 'div', onkeyup: function(){}, onfocusout:function(){}, onblur:function(){} };
var mce_validator = $("#mc-embedded-subscribe-form").validate(options);
$("#mc-embedded-subscribe-form").unbind('submit');//remove the validator so we can get into beforeSubmit on the ajaxform, which then calls the validator
options = { url: 'http://preseed.us3.list-manage1.com/subscribe/post-json?u=6d760d307e7d9e29e2344532b&id=96647ebc16&c=?', type: 'GET', dataType: 'json', contentType: "application/json; charset=utf-8",
beforeSubmit: function(){
$('#mce_tmp_error_msg').remove();
$('.datefield','#mc_embed_signup').each(
function(){
var txt = 'filled';
var fields = new Array();
var i = 0;
$(':text', this).each(
function(){
fields[i] = this;
i++;
});
$(':hidden', this).each(
function(){
var bday = false;
if (fields.length == 2){
bday = true;
fields[2] = {'value':1970};//trick birthdays into having years
}
if ( fields[0].value=='MM' && fields[1].value=='DD' && (fields[2].value=='YYYY' || (bday && fields[2].value==1970) ) ){
this.value = '';
} else if ( fields[0].value=='' && fields[1].value=='' && (fields[2].value=='' || (bday && fields[2].value==1970) ) ){
this.value = '';
} else {
if (/\[day\]/.test(fields[0].name)){
this.value = fields[1].value+'/'+fields[0].value+'/'+fields[2].value;	
} else {
this.value = fields[0].value+'/'+fields[1].value+'/'+fields[2].value;
}
}
});
});
$('.phonefield-us','#mc_embed_signup').each(
function(){
var fields = new Array();
var i = 0;
$(':text', this).each(
function(){
fields[i] = this;
i++;
});
$(':hidden', this).each(
function(){
if ( fields[0].value.length != 3 || fields[1].value.length!=3 || fields[2].value.length!=4 ){
this.value = '';
} else {
this.value = 'filled';
}
});
});
return mce_validator.form();
}, 
success: mce_success_cb
};
$('#mc-embedded-subscribe-form').ajaxForm(options);


});
}
function mce_success_cb(resp){
$('#mce-success-response').hide();
$('#mce-error-response').hide();
if (resp.result=="success"){
$('#mce-'+resp.result+'-response').show();
$('#mce-'+resp.result+'-response').html(resp.msg);
$('#mc-embedded-subscribe-form').each(function(){
this.reset();
});
} else {
var index = -1;
var msg;
try {
var parts = resp.msg.split(' - ',2);
if (parts[1]==undefined){
msg = resp.msg;
} else {
i = parseInt(parts[0]);
if (i.toString() == parts[0]){
index = parts[0];
msg = parts[1];
} else {
index = -1;
msg = resp.msg;
}
}
} catch(e){
index = -1;
msg = resp.msg;
}
try{
if (index== -1){
$('#mce-'+resp.result+'-response').show();
$('#mce-'+resp.result+'-response').html(msg); 
} else {
err_id = 'mce_tmp_error_msg';
html = '<div id="'+err_id+'" style="'+err_style+'"> '+msg+'</div>';

var input_id = '#mc_embed_signup';
var f = $(input_id);
if (ftypes[index]=='address'){
input_id = '#mce-'+fnames[index]+'-addr1';
f = $(input_id).parent().parent().get(0);
} else if (ftypes[index]=='date'){
input_id = '#mce-'+fnames[index]+'-month';
f = $(input_id).parent().parent().get(0);
} else {
input_id = '#mce-'+fnames[index];
f = $().parent(input_id).get(0);
}
if (f){
$(f).append(html);
$(input_id).focus();
} else {
$('#mce-'+resp.result+'-response').show();
$('#mce-'+resp.result+'-response').html(msg);
}
}
} catch(e){
$('#mce-'+resp.result+'-response').show();
$('#mce-'+resp.result+'-response').html(msg);
}
}
}

<!--codeends-->

lllbe/mediajs/onepage.js
/*-------------------------------------------------------------------------

	Theme Name: EGO - html - v.1
	
	For any questions concerning this theme please refer to documention or
	our forum at support.udfrance.com.

/*------------------------------------------------------------------------

//GENERAL FUNCTONS ///////////////////////////////////////////////////////

-------------------------------------------------------------------------*/

$(document).ready(function(){
	
	/*vars used throughout*/
	var wh,
		scrollSpeed = 1000,
		parallaxSpeedFactor = 0.6,
		scrollEase ='easeOutExpo',
		targetSection,
		sectionLink = 'a.navigateTo',
		menuLink = $('ul.navigation li a'),
	 	section = $('.section'),
		toggleMenu =$('.mobileMenuToggle'),
		foliothumb = $('.folio-thumb'),
		thumbW,
		thumbH,
		thumbCaption,
		target,
		hoverSpeed=500,
		hoverEase='easeOutExpo';


	//INIT --------------------------------------------------------------------------------/
	
	
	if(isMobile == true){
	
		$('.header').addClass('mobileHeader');	//add mobile header class
		
		//Clone header
		$('.header').clone().prependTo('.page');
		
		//Set section links
		$('.page').each(function() {
		 var curr_sec=$(this).attr('id');
		 $(this).children('.header').children('.inner').children('.navigation').children('li').each(function() {
		  if($(this).children('a').attr('href').substring(1)==curr_sec) {
		   	$(this).children('a').addClass('active');
		  }
		 });
		});
	
	}else{
		
		$('.page').addClass('desktop');
		$('.teaser').addClass('fixed');

	}
	
	
		
	
	
			
	//PARALLAX ----------------------------------------------------------------------------/
	
	
	$(window).bind('load', function() { 
								  
		parallaxInit();						  
		
	});

	function parallaxInit(){
		
		if(isMobile == true) return false;

		$('#teaser-1').parallax();
		$('#teaser-2').parallax();
		$('#teaser-3').parallax();		
		/*add as necessary*/
		
	}
	
	
	//HOMEPAGE SPECIFIC -----------------------------------------------------------------/
	
	
	function sliderHeight(){
		
		wh = $(window).height();
		$('#homepage').css({height:wh});
	
	}
	
	sliderHeight();	

	var lH = $('.logo-homepage').height();
	var	lW = $('.logo-homepage').width();
	
	$('.logo-homepage').hover(function() {
		if(isMobile == true) return false;
		$(this).animate({width:lH+50,height:lW+50,marginLeft:-((lH+50)/2),marginTop:-((lW+50)/2)},{queue:false});
	},
	function() {
		if(isMobile == true) return false;
		$(this).animate({width:lH,height:lW,marginLeft:-lH/2,marginTop:-lW/2},{queue:false});
	});

	//PAGE SPECIFIC ---------------------------------------------------------------------/
		
	
		/*page scrolling
		-------------------*/
		
		$(document).on('click', sectionLink, function(event) { 

			//kill slider timer
			$.fn.epicSlider.killTimer();

			//get current
			targetSection = $(this).attr('href');
			
			//Set doc title
			document.title = 'Preseed:' + ( targetSection.replace( /[_\-\#\!\.\/]/g, ' ' ));

			//prevent click if active
			//if($(this).hasClass('active')) return false;
				
			//get pos of target section
			var targetOffset = $(targetSection).offset().top+1;

			//scroll			 
			$('html,body').animate({scrollTop: targetOffset}, scrollSpeed, scrollEase, function() {

					//set hash
					//window.location.hash = targetSection

			 });
			
			return false;
			//event.preventDefault();
		
		});
		
		

		/*nav handling
		-------------------*/
		
		$(function(){	   
				   
			if(isMobile == true) return false;	   

			section.waypoint({
						  
				handler: function(event, direction) {
				
					var activeSection = $(this);
					
					if (direction === "up") {
						
						activeSection = activeSection.prev();
						
					}

					var activeMenuLink = $('ul.navigation li a[href=#' + activeSection.attr('id') + ']');
					
					menuLink.removeClass('active');
					activeMenuLink.addClass('active');
					
					
					//position header for mobile - solves fixed position issue
					var targetOffset = $(activeSection).offset().top;
					if(isMobile == true && $(window).scrollTop() >= wh) $('.header').css({top:targetOffset,display:'none'}).fadeIn();
					
		
				},
				
				offset: '35%'			//when it should switch on consecutive page
				
			});
	
		});
	
		/*nav reveal
		-------------------*/

		 $(window).bind('scroll', function(){
			
			   headerReveal();
			   
		}); 
	 
	 
		function headerReveal(){
			
			 if(isMobile == true) return false;	

			if ($(window).scrollTop() >= wh){
	 
					if(!$('.header').is(':animated')) {
						
						$('.header').stop(true,true).slideDown();
						/*push elements out of view when scrolling*/
						if(isMobile != true) $('.epic-graphic, .epic-caption,#epic-navigation').css({position:'absolute'});	  
					}
			}else{
					if(!$('.header').is(':animated')) {	
					
						$('.header').stop(true,true).slideUp();
						$('ul.navigation').fadeOut();
						toggleMenu.removeClass('open');
						if(isMobile != true) $('.epic-graphic, .epic-caption,#epic-navigation').css({position:'fixed'});
						
					}
								  
			} 
	
		 }

		
	//ROLLOVER SPECIFIC ---------------------------------------------------------------------/
		
	
		/*folio
		-------------------*/
		
		foliothumb.on({

		 mouseenter: function () {
	
	 
			 //check if device is mobile 
			 //or within an inactive filter category
			 //or if its video content in which case do nothing
			 if(isMobile == true) return false;
			 
			 thumbW = foliothumb.find('a').find('img').width();
			 thumbH = foliothumb.find('a').find('img').height();
			 
			//get refrences needed
		 	thumbCaption = $(this).find('a').attr('title');
			
			//add rolloverscreen
			if(!$(this).find('a').find('div').hasClass('folio-thumb-rollover')) $(this).find('a').append('<div class="folio-thumb-rollover"></div>');
			
			
			//set it to the image size and fade in
			hoverScreen = $('.folio-thumb-rollover')
			hoverScreen.css({width:thumbW,height:thumbH});

			
			//make sure caption is filled out
			if (typeof thumbCaption !== 'undefined' && thumbCaption !== false && $(this).find(hoverScreen).is(':empty')) {
				
				//construct rollover & animate
   				$(this).find(hoverScreen).append('<div class="thumbInfo">'+thumbCaption+'</div>');
				target = $(this).find(hoverScreen);
				target.stop().animate({opacity:1},hoverSpeed, hoverEase);
			}
			
		}, 
		
		  mouseleave: function () {
		
			if(isMobile == true) return false;
			
			//animate out
			$(this).find(hoverScreen).animate({opacity:0},hoverSpeed,'linear',function(){
	
					//delete rollover
				   $(this).remove();
				
			});
			
		
		 }
	
	  });
		
	
	
		
	 //WINDOW EVENTS ---------------------------------------------------------------------/	
	 
	 
	 $(window).bind('resize',function() {
		
		//Update slider height
	 	sliderHeight();
		
	});	
		
	
					
		

});